command/agent/consul/service_client.go

```diff
@@ -456,6 +456,12 @@ type ServiceClient struct {
 	allocRegistrations     map[string]*serviceregistration.AllocRegistration
 	allocRegistrationsLock sync.RWMutex
 
+	// serviceTokens is a map of service ID -> Consul tokens,
+	// where the token is set in the Workload by the client hooks when using
+	// Workload Identity. Access should be protected by allocRegistrationsLock
+	serviceTokens     map[string]string
```

Comment:
Would this be where the new `consul_hook` stores the tokens it generates?

Reply:
No, the `consul_hook` will write them to... hm, hook resources? disk? But then the `group_service_hook` and `service_hook` will add a map of service names to Consul tokens to their `RegisterWorkload` call. And then this PR is doing the work of taking those tokens from the `RegisterWorkload` args and sticking them into this map.